src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java

```diff
@@ -351,6 +359,7 @@ public KinesisClientLibConfiguration(String applicationName,
                                          long parentShardPollIntervalMillis,
                                          long shardSyncIntervalMillis,
                                          boolean cleanupTerminatedShardsBeforeExpiry,
+                                         boolean ignoreUnexpectedChildShards,
```

Comment:
You're changing a public constructor and adding a parameter in the middle of it.  This will break anyone who uses the constructor.  I would prefer to add a new constructor with the parameter.  The other option is not to have the parameter, and use the with/set operations to configure the feature.

At some point we will look into switching to a builder pattern for the configuration.